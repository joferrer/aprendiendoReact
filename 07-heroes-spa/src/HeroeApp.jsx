import { AuthProvider } from "./auth"
import { AppRouter } from "./routers/AppRouter"


export const HeroeApp = () => {
  return (
    <AuthProvider>
       <AppRouter/>
    </AuthProvider>
  )
}
