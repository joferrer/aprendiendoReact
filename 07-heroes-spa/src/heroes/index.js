
export * from './pages';
export * from './routes/HeroRoutes';
export * from './components';
