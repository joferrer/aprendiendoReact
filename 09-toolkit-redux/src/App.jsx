import { useDispatch, useSelector } from 'react-redux/es/exports'

import reactLogo from './assets/react.svg'
import './App.css'
import { decrement, increment, incrementByAmount } from './store/slices/counter';

function App() {
  
  const {counter} = useSelector(state => state.counter);

  const dispatch = useDispatch();

  return (
    <div className="App">
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src="/vite.svg" className="logo" alt="Vite logo" />
        </a>
        <a href="https://reactjs.org" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className="card">
        <button onClick={() => dispatch(increment())}>
          count is {counter}
        </button>
        <button onClick={() => dispatch(decrement())}>
          decrement
        </button>
        <button onClick={() => dispatch(incrementByAmount(2))}>
          increment by 2
        </button>
        
      </div>
      
    </div>
  )
}

export default App
