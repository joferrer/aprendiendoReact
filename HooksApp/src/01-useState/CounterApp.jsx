import { useState } from "react"

export const CounterApp = () => {


    const [counter, setcounter] = useState({
        counter1: 0,
        counter2: 0,
        counter3: 0
    });

    const {counter1,counter2,counter3} = counter;

    const addCounter = ()=>{
        setcounter({
            counter1: counter1+1,
            counter2: counter2 + 2,
            counter3: counter3 +3
        });

    }

    const plusOneCounter = ()=>{
        setcounter({
            ...counter,
            counter1: counter1+1,
            
        });
    };

    const plusTwoCounter = ()=>{
        setcounter({
            ...counter,
            counter2: counter2+2
        });
    };
    const plusThreeCounter = ()=>{
        setcounter({
            ...counter,
            counter3: counter3+3
            
        });
    };

  return (
    <>
        <h2>CounterApp</h2>
        <hr />

        <h3>Counter 1: {counter1}</h3>
        <h3>Counter 2: {counter2}</h3>
        <h3>Counter 3: {counter3}</h3>

        <button className="btn btn-outline-primary" onClick={addCounter}>+</button>
        <button className="btn btn-outline-primary" onClick={plusOneCounter}>+1</button>
        <button className="btn btn-outline-primary" onClick={plusTwoCounter}>+2</button>
        <button className="btn btn-outline-primary" onClick={plusThreeCounter}>+3</button>
    </>
  )
}
