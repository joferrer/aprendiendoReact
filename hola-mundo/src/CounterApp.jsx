import { useState } from 'react';
import PropTypes from 'prop-types'

export function CounterApp ({value}){

    const [[counter,nombre],setCounter] = useState([value,'Jeison']);



    const incrementar = ()=>{
        setCounter([counter + 1, nombre+' Ferrer']);
    };
    const restar = ()=>{
        setCounter([counter-1,nombre-' Ferrer']);
    };
    const reset=()=>{
        setCounter([value,'Jeison']);
    };

    return(<>
        <h1>CounterApp</h1>
        <h2>{counter}</h2>
        <h3>{nombre}</h3>
        <button onClick={incrementar}>Incrementar</button>
        <button onClick={restar}>Decrementar</button>
        <button onClick={reset}>Reset</button>
    
    </>);
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired

};
