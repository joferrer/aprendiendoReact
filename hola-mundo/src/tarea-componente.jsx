import './css/scale.css'
import PropTypes from 'prop-types';

const items = () => ['J', 'E','I','S','O','N'];

const itemsStylee = () =>{
    const lista = items().map((item)=>item+" P ");
    
    return lista;
};

export function Componente ({nombre}){
    
    return (
        
        <>
            <h1>{items().map((item,index)=><div key={"l"+index} className="scale" >{item+ " "}</div>)}</h1>
            <h2>{nombre}</h2>
             
        </>
    );
}

Componente.propTypes={
    nombre : PropTypes.string.isRequired
};

Componente.defaultProps = {
    nombre : "Desconocido"
}
