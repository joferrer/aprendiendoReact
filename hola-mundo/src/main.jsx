import React from 'react'
import {createRoot} from "react-dom/client"; // llamada al React DOM para renderizar componentes

import {App} from './App';



// react 18
const root = createRoot(document.getElementById('root'));
root.render(<React.StrictMode>
    <App/>
    
</React.StrictMode>);