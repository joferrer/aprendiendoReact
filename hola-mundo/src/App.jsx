import { CounterApp } from "./CounterApp";
import { Componente } from "./tarea-componente";
export function App(){
    return (
        <div>
            <h1>Hola mundo!!! </h1>
            <Componente  />
            <CounterApp value={0}/>
            </div>
    );
}