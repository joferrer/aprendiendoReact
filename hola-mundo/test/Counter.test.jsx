import {App} from '../src/App';
import {CounterApp} from '../src/CounterApp';
import {render, screen} from '@testing-library/react';


describe('Test para el contador', ()=>{

    const value = 100;

    test('Hacer match con el snapshot',()=>{

        const {container}=render(<CounterApp value={value}/>);
        expect(container).toMatchSnapshot();

    });


    test('Reset funcionando correctamente' , ()=>{

        render(<CounterApp value={value}/>);
        
        expect(screen.getByText(value));
        expect(screen.getByRole('heading',{ level: 2 } ).innerHTML).toEqual('100');

    });


});