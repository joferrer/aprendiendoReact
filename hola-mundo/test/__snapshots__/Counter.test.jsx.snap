// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test para el contador Hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <h3>
    Jeison
  </h3>
  <button>
    Incrementar
  </button>
  <button>
    Decrementar
  </button>
  <button>
    Reset
  </button>
</div>
`;
